src/celeritas/optical/CerenkovGenerator.hh

```diff
@@ -0,0 +1,158 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/CerenkovGenerator.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <cmath>
+
+#include "corecel/Assert.hh"
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/data/Collection.hh"
+#include "corecel/math/ArrayUtils.hh"
+#include "celeritas/grid/GenericXsCalculator.hh"
+#include "celeritas/grid/XsGridData.hh"
+#include "celeritas/random/distribution/BernoulliDistribution.hh"
+#include "celeritas/random/distribution/GenerateCanonical.hh"
+#include "celeritas/random/distribution/UniformRealDistribution.hh"
+
+#include "CerenkovDistribution.hh"
+#include "PhotonPrimary.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Sample Cerenkov photons.
+ */
+class CerenkovGenerator
+{
+  public:
+    // Placeholder struct for tabulated index of refraction values
+    // TODO: remove once we store optical properties
+    struct RefractiveIndex
+    {
+        using Values
+            = Collection<real_type, Ownership::const_reference, MemSpace::native>;
+
+        GenericGridData const& data;
+        Values const& reals;
+    };
+
+  public:
+    // Construct from optical properties and distribution parameters
+    inline CELER_FUNCTION
+    CerenkovGenerator(RefractiveIndex const& refractive_index,
```

Comment:
Yeah I agree, this is just a placeholder since we haven't implemented the optical properties data structures yet.  I imagine we would pass in the optical properties as an argument and use the material ID in the Cerenkov distribution to get the correct refractive index data.

Reply:
Are going to have one long-lived object that will be re-used for any material, or will we construct one generator every time we need to sample the photons within that material? If the latter, a refractive index in the constructor is already fine, and the `RefractiveIndex` data will just come from an `OpticalParamsView` or directly from `MaterialView`.

Reply:
Right now it's set up to be one generator per distribution, so the latter (but we can easily change that later if needed). So yes, either retrieving the optical data for the material inside the generator or passing it in in the constructor will work.